src/lib/components/tab_view.tsx

```diff
@@ -1,14 +1,27 @@
-import React from 'react'
-import { StyleSheet, View } from 'react-native'
+import * as React from 'react'
+import { StyleSheet, View, ViewProperties, TouchEventHandler } from 'react-native'
 
 import SwitchView from './switch_view'
 
-export default class TabView extends React.Component {
+interface Props extends ViewProperties {
+  titles: string[]
+  onSelectionChange: TouchEventHandler<SwitchView>
+  selectedIndex: number
+}
+
+export default class TabView extends React.Component<Props, {}> {
+  static propTypes = {
+    ...SwitchView.propTypes
+  }
+
   render() {
     const { children, ...props } = this.props
     return (
       <View>
-        <SwitchView style={styles.switch} {...props} />
```

Comment:
This spread operator gave me a lot of trouble - I did appreciate the elegance of this class though so if someone knows how to pass the props in this way without ts errors, it'll be very appreciated 😄 .